dbapi interface

remdragon opened the pull request:
I found sqlite3worker extremely helpful to me, and made a number of enhancements that I wanted to share back with the project:

1) dbapi compatible interface that implements enough features to be a drop-in replacement for 3 different database abstraction libraries that I use. This can be dropped by replacing "import sqlite3" with "import sqlite3worker as sqlite3"

2) use Queue objects in place of uuid tokens, this simplifies several code constructs including the back-off delay while waiting for a response.

3) eliminate auto-commit and support commit calls directly giving user complete control of how often they want to commit.

4) catch exceptions and throw in caller's thread

5) eliminate need for "thread_running" attribute by using threading.Thread.join()

6) Try to protect the user from accidentally opening two different worker objects on the same database.

Commit:
Updated email.

Commit:
Updated email.

Commit:
Ignore .clean files.

dashawn888 committed:
Small pep8.
Changed to event system vs double back timer. Thanks @apsc for showing me the way.
Explicitly close out the sqlite3 db object.
Couple more tests.

dashawn888 committed:
Wrong order

dashawn888 committed:
Updated many internals to private to clean up documentation and object exploration.

dashawn888 committed:
Updated max_queue_size to private.

dashawn888 committed:
Updated installation instructions. Updated email.

dashawn888 committed:
Added from https://github.com/github/gitignore/blob/master/Python.gitignore

dashawn888 committed:
Pinged version to 1.1.

dashawn888 committed:
Pinged version to 1.1.

dashawn888 committed:
Preping for pypy upload.

dashawn888 committed:
Added classifiers.

dashawn888 committed:
Updated docs.

dashawn888 committed:
Pinged version.

dashawn888 committed:
Adding travis tests.

dashawn888 committed:
Setting up travis auto build.

dashawn888 committed:
Adding repo badge.

dashawn888 committed:
Fixed bug breaking threadsafe by adding back an ugly sleep. Will come back to find more elegant way to fix. Created test to find this bug.

dashawn888 committed:
Pinged version.

dashawn888 committed:
Added locking for the closing call to prevent weird double close bug. Reduced thread count in thread tests.

dashawn888 committed:
To 1.1.4

Djaler committed:
Update __init__.py

Because it cause to error:

    __version__ = 1.1.3
                      ^
SyntaxError: invalid syntax

Djaler committed:
Update __init__.py

Because

File "/home/djaler/Projects/Python/TranslationBot/venv/lib/python3.5/site-packages/sqlite3worker/__init__.py", line 28, in <module>
    from sqlite3worker import Sqlite3Worker
ImportError: cannot import name 'Sqlite3Worker'

dashawn888 committed:
Ading wait for thread to die. This prevents a rare race condition were it's closing but not quite closed.

dashawn888 committed:
Updated to 1.1.5.

dashawn888 committed:
Merge branch 'master' into patch-1

dashawn888 committed:
Merge pull request #2 from Djaler/patch-2

Update __init__.py

dashawn888 committed:
Merge pull request #1 from Djaler/patch-1

Update __init__.py

dashawn888 committed:
Python2 vs Python3 compat.

dashawn888 committed:
Added python 3.7.

dashawn888 committed:
Added pypi docs.

dashawn888 committed:
Updated to 1.1.6 to get missing docs back in.

dashawn888 committed:
Removed all sleeps. Finished the thread event solution for syncronizing objects.

dashawn888 committed:
v 1.1.7

dashawn888 committed:
Moved the notification to closer to the actual execution of the query.

dashawn888 committed:
Removed intermediate variable.

remdragon committed:
Update sqlite3worker.py

palantirtech commented:
Thanks for your interest in [palantir/sqlite3worker](https://github.com/palantir/sqlite3worker), @remdragon! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

dashawn888 commented:
Hi,

Thank you for the updates. I have forked the project and put many updates into the fork. https://github.com/dashawn888/sqlite3worker and placed many updates in it. 

This is also where the pip points to. 

https://pypi.python.org/pypi/sqlite3worker/1.1.7

If you could please put your branch into the new source or I can put your updates in and make sure to give you credit. 

Thanks,

Shawn



Sent from my iPhone

> On Aug 10, 2017, at 6:54 PM, Royce Mitchell <notifications@github.com> wrote:
> 
> I found sqlite3worker extremely helpful to me, and made a number of enhancements that I wanted to share back with the project:
> 
> dbapi compatible interface that implements enough features to be a drop-in replacement for 3 different database abstraction libraries that I use. This can be dropped by replacing "import sqlite3" with "import sqlite3worker as sqlite3"
> 
> use Queue objects in place of uuid tokens, this simplifies several code constructs including the back-off delay while waiting for a response.
> 
> eliminate auto-commit and support commit calls directly giving user complete control of how often they want to commit.
> 
> catch exceptions and throw in caller's thread
> 
> eliminate need for "thread_running" attribute by using threading.Thread.join()
> 
> Try to protect the user from accidentally opening two different worker objects on the same database.
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/palantir/sqlite3worker/pull/9
> 
> Commit Summary
> 
> Update sqlite3worker.py
> File Changes
> 
> M sqlite3worker.py (497)
> Patch Links:
> 
> https://github.com/palantir/sqlite3worker/pull/9.patch
> https://github.com/palantir/sqlite3worker/pull/9.diff
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

remdragon commented:
I discovered your fork and have been trying to compare the code. I believe
I independently addressed all the differences ( in some way or another )
between the palantir version and your fork.

I'm afraid I'm a bit of a newb when it comes to git. Are you asking me to
just submit the same pull request to your fork that I did to palantir?



Royce Mitchell, IT Consultant
ITAS Solutions
royce3@itas-solutions.com

On Thu, Aug 10, 2017 at 9:04 PM, Shawn Lee <notifications@github.com> wrote:

> Hi,
>
> Thank you for the updates. I have forked the project and put many updates
> into the fork. https://github.com/dashawn888/sqlite3worker and placed
> many updates in it.
>
> This is also where the pip points to.
>
> https://pypi.python.org/pypi/sqlite3worker/1.1.7
>
> If you could please put your branch into the new source or I can put your
> updates in and make sure to give you credit.
>
> Thanks,
>
> Shawn
>
>
>
> Sent from my iPhone
>
> > On Aug 10, 2017, at 6:54 PM, Royce Mitchell <notifications@github.com>
> wrote:
> >
> > I found sqlite3worker extremely helpful to me, and made a number of
> enhancements that I wanted to share back with the project:
> >
> > dbapi compatible interface that implements enough features to be a
> drop-in replacement for 3 different database abstraction libraries that I
> use. This can be dropped by replacing "import sqlite3" with "import
> sqlite3worker as sqlite3"
> >
> > use Queue objects in place of uuid tokens, this simplifies several code
> constructs including the back-off delay while waiting for a response.
> >
> > eliminate auto-commit and support commit calls directly giving user
> complete control of how often they want to commit.
> >
> > catch exceptions and throw in caller's thread
> >
> > eliminate need for "thread_running" attribute by using
> threading.Thread.join()
> >
> > Try to protect the user from accidentally opening two different worker
> objects on the same database.
> >
> > You can view, comment on, or merge this pull request online at:
> >
> > https://github.com/palantir/sqlite3worker/pull/9
> >
> > Commit Summary
> >
> > Update sqlite3worker.py
> > File Changes
> >
> > M sqlite3worker.py (497)
> > Patch Links:
> >
> > https://github.com/palantir/sqlite3worker/pull/9.patch
> > https://github.com/palantir/sqlite3worker/pull/9.diff
> > —
> > You are receiving this because you are subscribed to this thread.
>
> > Reply to this email directly, view it on GitHub, or mute the thread.
> >
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/palantir/sqlite3worker/pull/9#issuecomment-321717764>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADka1yvcw8SibPY3izN_IYnx3TitKyTHks5sW7ajgaJpZM4O0K7o>
> .
>

remdragon committed:
fix tests to match changes in sqlite3worker

remdragon committed:
fixes from unit testing

renamed sqlite3_conn to _sqlite3_conn to match tests, fix bad LOGGER.debug() call,

remdragon committed:
py3 compatibility

remdragon committed:
py3 compatibility

remdragon committed:
Update sqlite3worker.py

remdragon committed:
Merge branch 'master' into master

remdragon committed:
pathlib is overkill here

remdragon committed:
sqlite3worker.py: refactor Sqlite3Worker class to share thread objects so that we can have multiple instances of a Sqlite3Worker. This fixes bug with multiple dbapi connections not shutting down the thread on each other. Replaced OperationalError with ProgrammingError where it seemed more appropriate.
sqlite3worker_test.py: wrote test to reproduce an error that occurred in production when opening multiple dbapi connections to the same database from different threads

remdragon committed:
sqlite3worker.py: fix bug in Sqlite3Worker.close()
sqlite3worker_test.py: it helps when you actually commit the new tests you write

remdragon committed:
sqlite3worker.py: replace sqlite3.Row with dict_factory because sqlite3.Row doesn't like our fake cursor, fix bug in Sqlite3WorkerExecuteScript, remove "no cover" pragmas that are now covered in testing, fix typo in Sqlite3Worker.commit(), dbapi interface now keeps and applies row_factory itself instead of relying on Sqlite3Worker to do it - this fixes bug where different connections have different row factories
sqlite3worker_test.py: add a bunch more tests to get code coverage to 100% and prevent regression of bugs fixed in sqlite3worker.py

remdragon committed:
sqlite3worker_test.py: py3 compatibility
